### ISSUE TITLE
False positive E0244 for const generic

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#4803]
## Environment

* **IntelliJ Rust plugin version:** 0.2.114.2820-193-nightly
* **Rust toolchain version:** 1.42.0-nightly (0de96d37f 2019-12-19) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2019.3 Ultimate Edition (IU-193.5233.102)
* **Operating system:** macOS 10.13.6

## Problem description

![image](https://user-images.githubusercontent.com/2539310/71563064-c19f6380-2a9a-11ea-9e26-3d71fdd6089a.png)

## Steps to reproduce
```rust
#![feature(const_generics)]

struct Table<const N: usize>([i32; N]);

impl<const N: usize> Table<N> {
    fn new() -> Table<N> {
        Table([0; N])
    }
}
```

**Comment by mchernyavsky** [comment]
Fixed in https://github.com/intellij-rust/intellij-rust/pull/7001.
